### PR TITLE
Replace deprecated code with newer APIs

### PR DESCRIPTION
**Pull request by Eitot**
I ran the static analyser with the deployment target set to macOS 12.3 to see if any deprecation warnings show up. These commits address some of those warnings (e.g. not the C-based Keychain APIs) by using newer APIs where available.

## Misc Checklist:

- [ ] My change requires a documentation update on [Sparkle's website repository](https://github.com/sparkle-project/sparkle-project.github.io)
- [ ] My change requires changes to generate_appcast, generate_keys, or sign_update

Only bug fixes to regressions or security fixes are being backported to the 1.x (master) branch now. If you believe your change is significant enough to backport, please also create a separate pull request against the master branch.

## Testing

I tested and verified my change by using one or multiple of these methods:

- [X] Sparkle Test App
- [X] Unit Tests
- [ ] My own app
- [x] Other (Clang-Analyser)

macOS version tested: 12.3.1

### CODE REVIEW

File: Sparkle/SUApplicationInfo.m [hunk 1]
```diff
@@ -12,6 +12,7 @@
 #import "SUBundleIcon.h"
 #import "SUHost.h"
 #import <AppKit/AppKit.h>
+#import <UniformTypeIdentifiers/UniformTypeIdentifiers.h>
```
Comment (zorgiepoo) [inline Sparkle/SUApplicationInfo.m:15]: UniformTypeIdentifiers may need to be added in linked libraries for Sparkle framework and Installer Progress targets and as `Optional` to make sure it's weak linked so it's fine on systems less than macOS 11.

**Comment by zorgiepoo** [comment]
Looks good besides one weak linkage concern. After that I will probably give it a basic test run on one of my VMs running < macOS 11.

File: Sparkle/SUApplicationInfo.m (hunk 1 above)
Comment (zorgiepoo) [reply Sparkle/SUApplicationInfo.m:15]: Alright Xcode already made this weak automatically so this is not an issue, never mind!

Comment (Eitot) [reply Sparkle/SUApplicationInfo.m:15]: I was under the impression that Clang does this automatically when CLANG_MODULES_AUTOLINK is enabled (which is the case for Sparkle and Installer Progress).

`otool` indicates this as well (if I interpret this correctly).

Comment (zorgiepoo) [reply Sparkle/SUApplicationInfo.m:15]: Yep you're right.

**Comment by zorgiepoo** [comment]
Merge conflict from previous changes.

**Comment by zorgiepoo** [comment]
Gave a run on macOS 10.12, seems good, thanks!

**Comment by zorgiepoo** [comment]
(The deprecated keychain APIs are only used for DSA keys btw which we've deprecated ourselves).

**Pull request merged**